src/realm/object-store/impl/realm_coordinator.cpp

```diff
@@ -649,11 +639,17 @@ RealmCoordinator::~RealmCoordinator()
             }
         }
     }
+
+    if (m_db) {
+        m_db->remove_commit_listener(this);
+    }
+
     // Waits for the worker thread to join
-    m_notifier = nullptr;
+    m_notifier.reset();
 
-    // Ensure the notifiers aren't holding on to Transactions after we destroy
-    // the History object the DB depends on
+    // If there's any active NotificationTokens they'll keep the notifiers alive,
+    // so tell the notifiers to release their Transactions so that the DB can
+    // be closed immediately.
```

Comment:
This comment was stale and the reason why we originally needed release_data() no longer applies, but it is still required for other reasons.